AWSAuthSDK/Sources/AWSUserPoolsSignIn/UserPoolsUI/AWSUserPoolNewPasswordRequiredViewController.m

```diff
@@ -75,9 +77,14 @@ - (void)setUpBackground {
 - (IBAction)onSignIn:(id)sender {
     NSString *password = [self.tableDelegate getValueForCell:self.passwordRow
                                                 forTableView:self.tableView];
-    
+    NSMutableDictionary *filteredDictionary = [[NSMutableDictionary alloc] initWithCapacity:self.passwordRequiredInput.userAttributes.count];
+    [self.passwordRequiredInput.userAttributes enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key, NSString * _Nonnull obj, BOOL * _Nonnull stop) {
```

Comment:
More efficient to just do a [`-[removeObjectForKey:]`](https://developer.apple.com/documentation/foundation/nsmutabledictionary/1416518-removeobjectforkey), which does nothing if the key is not present:

```objective-c
NSMutableDictionary *filteredDictionary = [self. passwordRequiredInput.userAttributes mutableCopy];
[filteredDictionary removeObjectForKey:AWSUserPool_EmailVerifiedKey];
```